LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceCache.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer
+{
+    using LoRaTools.LoRaMessage;
+    using LoRaTools.Utils;
+    using Microsoft.Extensions.Logging;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class LoRaDeviceCache : IDisposable
+    {
+        private readonly LoRaDeviceCacheOptions options;
+        private readonly ConcurrentDictionary<string, ConcurrentDictionary<string, LoRaDevice>> devAddrCache = new();
+        private readonly ConcurrentDictionary<string, LoRaDevice> euiCache = new();
+        private readonly object syncLock = new object();
+        private readonly NetworkServerConfiguration configuration;
+        private readonly ILogger<LoRaDeviceCache> logger;
+        private CancellationTokenSource? ctsDispose;
+        private readonly StatisticsTracker statisticsTracker = new StatisticsTracker();
+
+        protected LoRaDeviceCache(LoRaDeviceCacheOptions options, NetworkServerConfiguration configuration, ILogger<LoRaDeviceCache> logger, CancellationToken externalRefreshCancellationToken)
+        {
+            this.options = options;
+            this.ctsDispose = externalRefreshCancellationToken.CanBeCanceled ? CancellationTokenSource.CreateLinkedTokenSource(externalRefreshCancellationToken) : new CancellationTokenSource();
```

Comment:
What's the purpose of creating a linked token source when the only user of this protected constructor is actually always passing a CancellationToken.None?